ERCS/erc-7652.md

```diff
@@ -0,0 +1,160 @@
+---
+eip: 7652
+title: ERC-721 Guarantee Extension
+description: ERC-721 Guarantor role interface extension
+author: Liu.C.Dao (@CDao) <iunknow@163.com>, Sam <1047180870@qq.com>
+discussions-to: https://ethereum-magicians.org/t/erc-7652-eip-721-guarantee-extension/19284
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-03-10
+requires: 165, 721
+---
+
+## Abstract
+
+This specification defines functions outlining a guarantor role for instance of [EIP-721](./eip-721.md). The guarantee interface implements the user-set valuation and guarantee share for a given NFT (token ID), as well as the guarantee rights enjoyed and obligations assumed during subsequent transactions. An implementation enables the user to read or set the current guarantee value for a given NFT (token ID), and also realizes the distribution of guarantee interest and the performance of guarantee obligations. It sends the standardized events when the status changes. This proposal relies on and extends the existing [EIP-721](./eip-721.md).
+
+## Motivation
+
+NFT (token ID) commonly face the issue of insufficient market liquidity: the main reason being the lack of transparency in NFT pricing, making it difficult for users to cash out after trading and purchasing NFT (token ID).
+
+With the introduction of the guarantor role, different guarantor groups can offer various price guarantees for NFT (token ID), establishing a multi-faceted price evaluation system for NFT (token ID).
+
+After purchasing an NFT (token ID), users can return it to the guarantor at any time at the highest guaranteed price to protect their interests.
+
+Additionally, after fulfilling their guarantee obligations, the guarantor can also request subsequent guarantors to provide guarantee obligations.
+
+When an NFT (token ID) is owned by the guarantor, and since the guarantor can be a DAO organization, this expansion allows the NFT (token ID) to continue operating as a DAO, thus further enhancing the social or community recognition of the NFT (token ID).
+
+
+## Specification
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Every contract compliant to the `ERC721Guarantee` MUST implement the `IERC721Guarantee` guarantee interface.
```

Comment:
```suggestion
Every contract compliant with this proposal MUST implement the `IERC721Guarantee` guarantee interface.
```

Reply:
Thank you for your suggestion, the new version has made the relevant revisions to the wording.